src/dotnet/Audio.Service/AudioProcessor.cs

```diff
@@ -84,66 +84,66 @@ public AudioProcessor(Options settings, IServiceProvider services)
             await openSegment.Audio.WhenDurationAvailable.ConfigureAwait(false);
+            // close open audio segment when the duration become available
+            openSegment.Close(openSegment.Audio.Duration);
+            var closedSegment = await openSegment.ClosedSegmentTask.ConfigureAwait(false);
+            // we don't use cancellationToken there because we should finalize audio entry
+            // if it has been created successfully no matter of method cancellation
+            var audioBlobId = await AudioSegmentSaver.Save(closedSegment, CancellationToken.None)
+                .ConfigureAwait(false);
+            // this should already have been completed by this time
+            var audioEntry = await audioEntryTask.ConfigureAwait(false);
+            await FinalizeAudioEntry(openSegment, audioEntry, audioBlobId, CancellationToken.None)
+                .ConfigureAwait(false);
 
-            async Task FinalizeAudioProcessing()
-            {
-                // TODO(AY): We should make sure finalization happens no matter what (later)!
-                // TODO(AK): Compensate failures during audio entry creation or saving audio blob (later)
-                // real-time audio transmission has been completed
-                await publishAudioTask.ConfigureAwait(false);
-                // this should already have been completed by this time
-                var audioEntry = await audioEntryTask.ConfigureAwait(false);
-                // close open audio segment when the duration become available
-                await openSegment.Audio.WhenDurationAvailable.ConfigureAwait(false);
-                openSegment.Close(openSegment.Audio.Duration);
-                var closedSegment = await openSegment.ClosedSegmentTask.ConfigureAwait(false);
-                // we don't use cancellationToken there because we should finalize audio entry
-                // if it has been created successfully no matter of method cancellation
-                var audioBlobId = await AudioSegmentSaver.Save(closedSegment, CancellationToken.None)
-                    .ConfigureAwait(false);
-                await FinalizeAudioEntry(openSegment, audioEntry, audioBlobId, CancellationToken.None)
-                    .ConfigureAwait(false);
-
-                // we don't care much about transcription errors - basically we should finalize audio entry before
-                await transcribeTask.ConfigureAwait(false);
-            }
+            // we don't care much about transcription errors - basically we should finalize audio entry before
+            await transcribeTask.ConfigureAwait(false);
         }
         catch (Exception e) when (e is not OperationCanceledException) {
             Log.LogError(e, "Error processing audio stream {StreamId}", streamId);
             throw;
         }
+        catch (Exception e) {
+            Log.LogWarning(e, "Cancelled processing audio stream {StreamId}", streamId);
```

Comment:
I usually use "failed" phrasing in logging - i.e. `"Audio processing failed for stream #{StreamId}"`.

Reply:
We usually do not write cancellation errors  - and here I did that for debugging purposes - so it's not a failure